OpenEdXMobile/src/main/java/org/edx/mobile/googlecast/GoogleCastDelegate.java

```diff
@@ -0,0 +1,249 @@
+package org.edx.mobile.googlecast;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Handler;
+import android.view.MenuItem;
+
+import com.google.android.gms.cast.MediaInfo;
+import com.google.android.gms.cast.MediaLoadOptions;
+import com.google.android.gms.cast.MediaMetadata;
+import com.google.android.gms.cast.MediaStatus;
+import com.google.android.gms.cast.framework.CastContext;
+import com.google.android.gms.cast.framework.CastSession;
+import com.google.android.gms.cast.framework.CastState;
+import com.google.android.gms.cast.framework.CastStateListener;
+import com.google.android.gms.cast.framework.IntroductoryOverlay;
+import com.google.android.gms.cast.framework.SessionManagerListener;
+import com.google.android.gms.cast.framework.media.RemoteMediaClient;
+import com.google.android.gms.common.images.WebImage;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.MainApplication;
+import org.edx.mobile.core.IEdxEnvironment;
+import org.edx.mobile.model.db.DownloadEntry;
+import org.edx.mobile.module.analytics.Analytics;
+
+public class GoogleCastDelegate extends RemoteMediaClient.Callback implements
+        SessionManagerListener<CastSession> {
+
+    public interface CastSessionListener {
+        void onApplicationConnected();
+
+        void onApplicationDisconnected();
+
+        void onVideoComplete();
+
+        void onVideoPlaying();
+    }
+
+    private static GoogleCastDelegate delegate;
+    private final Context context;
+    private CastContext castContext;
+    private CastSession castSession;
+
+    private IntroductoryOverlay introductoryOverlay;
+    private CastSessionListener sessionListener;
+
+    private IEdxEnvironment environment;
+
+    public static GoogleCastDelegate instance() {
```

Comment:
It should be synchronized as per singleton pattern recommendation.

Reply:
It's still pending, let's make it a thread-safe singleton pattern
`    public static synchronized GoogleCastDelegate getInstance() {`